### DISCUSSION TITLE
Help needed building a custom theme

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [JanDeDobbeleer/oh-my-posh#1114] [Troubleshoot]
This is my current theme: https://gist.github.com/TheOnlyTails/8bbc4a2ff47ecd9cdcfb0d92dbca2e5e 
![image](https://user-images.githubusercontent.com/65342367/138765087-9483b696-2815-4ddb-a6d2-07d407dcd888.png)

I am trying to create something like so:
1. As in the current theme, there is a shell segment with a rounded start and "pointy" end.
2. If the terminal is in Admin mode, a root segment is displayed, similar to the following image, but with a "space" after it.
![image](https://user-images.githubusercontent.com/65342367/138765790-eeb86cf1-2db5-4916-8a0d-1d72e57b7f7e.png)
3. Then, the following segments, all with spaces between them, a "hollow pointy" start and a "pointy" end:
    - Path
    - Git
    - Java
    - Node.js
    - Python
    - etc.

4.  At last, a "rounding" segment, which would inherit the color of the previous segment and end on a rounder character, though I already have that implemented in the current theme.

Could you please help me here?

**Comment by JanDeDobbeleer** [comment]
@TheOnlyTails can you make this a little bit more explicit (and in show what you currently have created and would like to see removed/changed?) We can for sure help out.

**Reply by TheOnlyTails** [reply]
I've made a few improvements:
https://gist.github.com/TheOnlyTails/bdb2e379acf185f2d8f065b1c7b90dd7
![image](https://user-images.githubusercontent.com/65342367/139131027-40281bda-e5b5-4d15-b264-b6e8b3113a6f.png)

Basically, I want the same thing as this:
- first segment rounded with pointy end
- all other segments with inset pointy start (i. e. the start of the path segment in the picture) and pointy end, but with _spaces between them_ (this is the part I need help with)
- ending with a color-inheriting "rounder"

sort of like this picture, but with blank spaces (like the one between the 1st and 2nd segments in the picture) where there are white lines
![image](https://user-images.githubusercontent.com/65342367/139131571-ba4e14c5-f269-4a1b-8d92-31ea918b62ab.png)

hopefully that would clear things up

**Reply by JanDeDobbeleer** [reply]
@TheOnlyTails perfectly clear, thanks.

So basically, what you need to use is a diamond segment like this:

```json
"style": "diamond",
"leading_diamond": "<transparent,#ffeb3b></>",
"trailing_diamond": "",
```

You set the foreground to transparent and the background to the color of the segment. The question is if you want the last elements to also have a space in between and ALWAYS end with a rounded part regardless of which last segment?

<img width="666" alt="image" src="https://user-images.githubusercontent.com/2492783/139233940-38d8e17b-4519-4df6-be8f-d77ceffabd3a.png">

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "final_space": false,
  "console_title": true,
  "console_title_style": "template",
  "console_title_template": "{{.Host}} 📁 {{.Folder}}",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "vertical_offset": 1,
      "newline": true,
      "segments": [
        {
          "type": "shell",
          "style": "diamond",
          "foreground": "#c59b9b",
          "background": "#FF7700",
          "leading_diamond": "",
          "trailing_diamond": ""
        },
        {
          "type": "root",
          "style": "diamond",
          "leading_diamond": "<transparent,#ffff66></>",
          "trailing_diamond": "",
          "foreground": "#111111",
          "background": "#ffff66",
          "properties": { "root_icon": "" }
        },
        {
          "type": "path",
          "style": "diamond",
          "leading_diamond": "<transparent,#61AFEF></>",
          "trailing_diamond": "",
          "foreground": "transparent",
          "background": "#61AFEF",
          "properties": {
            "folder_separator_icon": "<transparent>  </>",
            "prefix": "  ",
            "style": "agnoster_full",
            "enable_hyperlink": true,
            "postfix": " "
          }
        },
        {
          "type": "git",
          "style": "diamond",
          "leading_diamond": "<transparent,#ffeb3b></>",
          "trailing_diamond": "",
          "foreground": "transparent",
          "background": "#ffeb3b",
          "properties": {
            "branch_icon": " ",
            "branch_identical_icon": "",
            "branch_gone_icon": "",
            "display_status": true,
            "display_stash_count": true,
            "display_upstream_icon": true
          }
        },
        {
          "type": "java",
          "style": "powerline",
          "powerline_symbol": "",
          "foreground": "#ffffff",
          "background": "#4063D8",
          "properties": {
            "prefix": "  ",
            "postfix": "",
            "missing_command_text": "Couldn't find Java version"
          }
        },
        {
          "type": "node",
          "style": "powerline",
          "powerline_symbol": "",
          "foreground": "#ffffff",
          "background": "#6CA35E",
          "properties": { "prefix": "  ", "postfix": "" }
        },
        {
          "type": "python",
          "style": "powerline",
          "powerline_symbol": "",
          "foreground": "#100e23",
          "background": "#906cff",
          "properties": { "prefix": "  ", "postfix": "" }
        },
        {
          "type": "text",
          "style": "diamond",
          "background": "inherit",
          "foreground": "inherit",
          "trailing_diamond": "",
          "properties": { "prefix": "", "text": "" }
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "type": "exit",
          "style": "plain",
          "foreground": "#d9d9d9",
          "properties": {
            "prefix": "❯",
            "always_enabled": true,
            "error_color": "#FF7700",
            "display_exit_code": false
          }
        }
      ]
    }
  ]
}
```

**Reply by TheOnlyTails** [reply]
> The question is if you want the last elements to also have a space in between and ALWAYS end with a rounded part regardless of which last segment?

@JanDeDobbeleer Yes, I do.
